homeassistant/components/zha/__init__.py

```diff
@@ -14,44 +15,47 @@
 from . import api
 from .core import ZHAGateway
 from .core.const import (
+    BAUD_RATES,
     COMPONENTS,
     CONF_BAUDRATE,
     CONF_DATABASE,
     CONF_DEVICE_CONFIG,
     CONF_ENABLE_QUIRKS,
     CONF_RADIO_TYPE,
     CONF_USB_PATH,
+    CONF_ZIGPY,
     DATA_ZHA,
     DATA_ZHA_CONFIG,
     DATA_ZHA_DISPATCHERS,
     DATA_ZHA_GATEWAY,
     DATA_ZHA_PLATFORM_LOADED,
-    DEFAULT_BAUDRATE,
-    DEFAULT_RADIO_TYPE,
     DOMAIN,
     SIGNAL_ADD_ENTITIES,
     RadioType,
 )
 from .core.discovery import GROUP_PROBE
 
 DEVICE_CONFIG_SCHEMA_ENTRY = vol.Schema({vol.Optional(ha_const.CONF_TYPE): cv.string})
-
+ZHA_CONFIG_SCHEMA = {
+    vol.Optional(CONF_BAUDRATE): cv.positive_int,
+    vol.Optional(CONF_DATABASE): cv.string,
+    vol.Optional(CONF_DEVICE_CONFIG, default={}): vol.Schema(
+        {cv.string: DEVICE_CONFIG_SCHEMA_ENTRY}
+    ),
+    vol.Optional(CONF_ENABLE_QUIRKS, default=True): cv.boolean,
+    vol.Optional(CONF_ZIGPY): dict,
+}
 CONFIG_SCHEMA = vol.Schema(
     {
         DOMAIN: vol.Schema(
-            {
-                vol.Optional(CONF_RADIO_TYPE, default=DEFAULT_RADIO_TYPE): cv.enum(
-                    RadioType
-                ),
-                CONF_USB_PATH: cv.string,
-                vol.Optional(CONF_BAUDRATE, default=DEFAULT_BAUDRATE): cv.positive_int,
-                vol.Optional(CONF_DATABASE): cv.string,
-                vol.Optional(CONF_DEVICE_CONFIG, default={}): vol.Schema(
-                    {cv.string: DEVICE_CONFIG_SCHEMA_ENTRY}
-                ),
-                vol.Optional(CONF_ENABLE_QUIRKS, default=True): cv.boolean,
-            }
-        )
+            vol.All(
+                cv.deprecated(CONF_USB_PATH, invalidation_version="0.112"),
+                cv.deprecated(CONF_BAUDRATE, invalidation_version="0.112"),
+                cv.deprecated(CONF_RADIO_TYPE, invalidation_version="0.112"),
+                ZHA_CONFIG_SCHEMA,
+            ),
+            extra=vol.ALLOW_EXTRA,
```

Comment:
We don't want to allow extra here in the domain schema, right?

Reply:
Yeah, I think we can remove it.